### PR TITLE
Course Settings - teacher remove - fixed delete arg, update to courselist on delete current teacher

### PR DESCRIPTION
**Pull request by darrenmason**
image

### CODE REVIEW

File: src/components/course-settings/remove-teacher.cjsx [hunk 1]
```diff
@@ -25,10 +24,11 @@ module.exports = React.createClass
       .value()
 
   goToDashboard: ->
-    @context.router.transitionTo('dashboard', {courseId: @props.courseId})
+    TeacherRosterStore.once 'deleted', ->
+      window.location.href = '/dashboard/'
```
Comment (philschatz) [inline src/components/course-settings/remove-teacher.cjsx]: Does `@context.router.transitionTo('dashboard', {courseId: @props.courseId})` no longer work in `TeacherRosterStore.once 'deleted', =>` ?

Comment (darrenmason) [inline src/components/course-settings/remove-teacher.cjsx]: When using transitionTo I encountered a SecurityTransgression error b/c the logged in course role had been deleted.  The combination of using once 'deleted' and window.location bypasses any missing course role errors and the courselist is always updated when reaching /dashboard/

Comment (philschatz) [inline src/components/course-settings/remove-teacher.cjsx]: hmm, does the following not work (combining `transitionTo` without the course id but wrapped in a `.once 'deleted'`)?

```coffee
TeacherRosterStore.once 'deleted', =>
  @context.router.transitionTo('dashboard')
```

Comment (philschatz) [inline src/components/course-settings/remove-teacher.cjsx]: Oh also, should the delete button in the modal be an `AsyncButton` so the user sees if the delete failed?

**Pull request merged**